### ISSUE TITLE
Apply payments to oldest outstanding transaction

### ISSUE THREAD

**Issue by SteveTeece** [microrealestate/microrealestate#126]
When applying payments from tenants, it would be beneficial to take a bulk payment (more than one period) and automatically apply the payment to the oldest outstanding transaction.  Also possibly include a check box to turn this feature on and off.

**Comment by camelaissani** [comment]
Can you describe the UI you would expect for this feature please?

**Comment by SteveTeece** [comment]
I don't think this needs much from the UI... just a check box to give the user a choice of applying the payment to the oldest outstanding invoices first.

For the logic, if the check box is filled in, take the payment amount, and apply it to the oldest outstanding invoice first. Any payment left over then gets applied to the next oldest invoice and so on.

If the check box is clear then current logic applies.

**Comment by camelaissani** [comment]
Accounting wise how do you make the link between the different amounts applied each month and the real payment by cheque, card?

**Comment by SteveTeece** [comment]
The accounting here is not complex, and probably easiest to describe in an example:

A tenant pays say $300 per week rent.  An invoice is raised each week.  The tenant has not paid rent for 3 weeks, then comes and gives a cheque for $1000.00.

The payment should be disbursed thus:
- $300 to oldest invoice - 3 weeks old (payment remaining to be disbursed $700)
- $300 to next oldest invoice - 2 weeks old (payment remaining $400)
- $300 to next oldest invoice - 1 week old (payment remaining $100)
- $100 kept in credit to be applied to the next invoice generated

As long as the total credit (amount paid) is equal to (==) the total debit (without getting into Charts of Accounts etc) the accounting should just work.,
